test_input.go

```diff
@@ -21,19 +21,22 @@ func NewTestInput() (i *TestInput) {
 }
 
 func (i *TestInput) Read(data []byte) (int, error) {
-	buf := <-i.data
-
-	var header []byte
-
-	if !i.skipHeader {
-		header = payloadHeader(RequestPayload, uuid(), time.Now().UnixNano(), -1)
-		copy(data[0:len(header)], header)
-		copy(data[len(header):], buf)
-	} else {
-		copy(data, buf)
+	select {
+	case buf := <-i.data:
+		var header []byte
+
+		if !i.skipHeader {
+			header = payloadHeader(RequestPayload, uuid(), time.Now().UnixNano(), -1)
+			copy(data[0:len(header)], header)
+			copy(data[len(header):], buf)
+		} else {
+			copy(data, buf)
+		}
+
+		return len(buf) + len(header), nil
+	case <-time.After(10* time.Second):
+		return 0, nil
```

Comment:
is it correct case or we should throw error here?

Reply:
Done